arch/arc/kernel/mcount.S

```diff
@@ -9,7 +9,12 @@
 #include <linux/init.h>
 #include <linux/linkage.h>
 #include <asm-generic/export.h>
+
+#ifdef CONFIG_ISA_ARCV3
```

Comment:
I failed to build kernel for ARCv3 HS5x target due to build errors:
```
  AS      arch/arc/kernel/mcount.o
/home/pvk/snps/arc/src/develop/linux/arch/arc/kernel/mcount.S: Assembler messages:
/home/pvk/snps/arc/src/develop/linux/arch/arc/kernel/mcount.S:56: Error: opcode 'pushl' not supported for target hs5x
/home/pvk/snps/arc/src/develop/linux/arch/arc/kernel/mcount.S:37:  Info: macro invoked from here
/home/pvk/snps/arc/src/develop/linux/arch/arc/kernel/mcount.S:51:   Info: macro invoked from here
/home/pvk/snps/arc/src/develop/linux/arch/arc/kernel/mcount.S:56: Error: opcode 'pushl' not supported for target hs5x
/home/pvk/snps/arc/src/develop/linux/arch/arc/kernel/mcount.S:38:  Info: macro invoked from here
/home/pvk/snps/arc/src/develop/linux/arch/arc/kernel/mcount.S:51:   Info: macro invoked from here
...
```
HS5x is ARCv3 but it also 32-bit. Additional condition is required:
```c
#if defined(CONFIG_ISA_ARCV3) && defined(CONFIG_64BIT)
#include <asm/asm-macro-64-bit.h>
#else
#include <asm/asm-macro-32-bit.h>                                                                                                                                                                                                                                                       #endif  
```

Reply:
Have you tested ftrace on 32-bit ARCv3? I'm receiving Oops and system freeze in attempt to set function_graph
```
# uname -a
Linux buildroot 5.15.127-714400-g963db12a58bf-dirty #3 PREEMPT Tue Nov 7 20:26:42 +04 2023 arc32 GNU/Linux
# mount -t tracefs nodev /sys/kernel/tracing
# cd /sys/kernel/tracing/
# cat available_tracers
function_graph function nop
# echo function_graph > current_tracer 

Oops
```

Reply:
Hello Pavel, sorry I neglected to state the ARCv3 version was only for 64 bits.
I have not tested for 32-bit ARCv3, only hs6x and I will check if I should either add the 32 bit version or block it from being used.

Reply:
@pavelvkozlov  added the "support" for hs5x :)
Thanks once again for pointing it out

Reply:
Checked by kselftests. Unfortunately, without dynamic ftrace and dynamic events vast majority of tests have been skipped, but anyway it is a good result for basic support.
```
# # of passed:  36
# # of failed:  0
# # of unresolved:  5
# # of untested:  0
# # of unsupported:  72
# # of xfailed:  0
# # of undefined(test bug):  0
ok 1 selftests: ftrace: ftracetest
```

Reply:
Several comments about patch:

- from the upstreaming point of view, it will be much better to have first patch with ftace support for ARCv2 and then extend it for ARCv3;
- please follow code standard and use tabs in mcount.S as it is in other source files;
- don't forget about blank end line;

Reply:
Done!